tests/test_maps.py

```diff
@@ -0,0 +1,170 @@
+import os
+import unittest
+from flask import jsonify
+from api import db
+from flask import Flask, request
+from api.models import Map
+from api.models import POI
+import requests
+from flask import jsonify
+import json
+
+# json objects to be tested
+map_add_json_1 = {
+    "map_year" : 1669,
+    "image_url": 'http://www.arizona-leisure.com/gfx/maps/valley-sun-map-760.gif'
+ }
+
+map_add_json_2 = {
+    "map_year" : 1670,
+    "image_url": 'http://www.fultonranchtownecenter.com/Post/sections/11/Images/FR_MasterPlan.jpg',
+}
+
+map_add_json_3 = {
+    "map_year" : 3000,
+    "image_url" : "https://pcavote.files.wordpress.com/2015/12/jonas-brothers-reunion.jpg"
+}
+
+map_add_json_bad_1 = {
+    "image_url": 'http://www.fultonranchtownecenter.com/Post/sections/11/Images/FR_MasterPlan.jpg'
+}
+
+map_add_json_bad_2 = {
+    "map_year": 2000
+}
+
+map_add_json_bad_3 = {
+
+}
+
+poi_add_json_1 = {
+    "name" : "The battle of the Bulge",
+    "date" : "5/20",
+    "description" : "people died",
+    "map_year" : 3000,
+    "x_coord" : "5",
+    "y_coord" : "7",
+    'links': [],
+    'media': [],
+    'story_ids': []
+}
+
+poi_add_json_2 = {
+    "name" : "The Great Herbal Fire",
+    "date" : "4/20",
+    "description" : "An uplifting experience for many people",
+    "map_year" : 3000,
+    "x_coord" : "7",
+    "y_coord" : "5",
+    'links': [],
+    'media': [],
+    'story_ids': []
+}
+
+poi_add_json_3 = {
+    "name" : "Nithin Becomes a Man",
+    "date" : "2/4",
+    "description" : "Nothing really",
+    "map_year" : 3000,
+    "x_coord" : "8",
+    "y_coord" : "9",
+    'links': [],
+    'media': [],
+    'story_ids': []
+}
+
+def find_map_occurence(map_obj):
+    data = requests.get('http://127.0.0.1:5000/maps')
+    map_year = map_obj['map_year']
+    map_year_list = list(filter(lambda x: x['map_year'], data.json()['result']['maps']))
+    while map_year in map_year_list:
+        map_year = random.randInt(1800,2018)
```

Comment:
you shouldn't use magic numbers. i think a better way is to start at an arbitrary year and keep incrementing until the year doesn't already exist